Silex

mathewpeterson opened the pull request:
This pull request starts the process to migrate from legacy to a modern stack. We will be using Silex at the moment and then to full stack Symfony. The purpose of this migration is to give the code the ability to apply true software engineering principles: SOLID, DRY, KISS, etc. We want the ability to decouple the business and presentation layer into their own.

mathewpeterson committed:
Adds Vagrant files

mathewpeterson committed:
Moves app files to web sub directory web

mathewpeterson committed:
Updates gitignore

mathewpeterson committed:
Adds memcache to puppet config

mathewpeterson committed:
Moves db config to dist file.

mathewpeterson committed:
Adds composer to project with a silex dependency.

abcrawford commented:
@mathewpeterson, 
Looks like you are still working on this. If you recommend any good resources on this framework so I can get my head around how it works, that would be great. Thanks for your work.

mathewpeterson commented:
@alleghenycreative http://symfony.com/doc/current/book/http_fundamentals.html is probably first read you should do. Basically, we need to get away from using echo's, includes, and multiple endpoints (index.php, logout.php, etc) and create a single entry and single exit strategy.

All though we are using Silex, it still uses Symfony components.

mathewpeterson committed:
Fixes php notices

mathewpeterson committed:
Adds memcache to puppet config

mathewpeterson committed:
Disables error reporting.

mathewpeterson committed:
Updates composer to include new deps and adds project meta data

mathewpeterson committed:
Adjusts whitespace.

mathewpeterson committed:
Implements Silex.

Adds Application Router
Adds .htaccess (though, in puppet config for now)
Adds LegacyController.

All routes are now going through the Silex front-end controller.

mathewpeterson committed:
Adds composer alias

mathewpeterson commented:
@alleghenycreative I apologize for the lengthy commit diff. I promise my future pull requests will be much shorter.
